diesel/src/connection/statement_cache.rs

```diff
@@ -189,9 +189,27 @@ where
         T: QueryFragment<DB> + QueryId,
         F: FnOnce(&str, PrepareForCache) -> QueryResult<Statement>,
```

Comment:
I think it should be fine to take a `&mut dyn Fn()` here already. It might be also fine to just change the API of `cached_statement` to what `cached_statement_inner` now looks like. This API is only public behind the "unstable" `i-implement-a-third-party-backend-and-opt-into-breaking-changes` feature flag.

Reply:
Sounds good, done. I went ahead and collapsed this extra function call (so `cached_statement_inner` no longer exists)